### PR TITLE
Updated Integration Collection Grid

### PR DESCRIPTION
**Pull request by sahilsuman933**
Signed-off-by: Sahil Suman <sahilsuman933@gmail.com>

**Description**

- Replaced Integration Grid with honeycomb-grid npm package
- Simplified logic for filtering Integration Collection
- Replaced the related integration from the Individual Integration page with honeycomb-grid

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit cbe3a37645bf3327e391229d453208bc2358529f at: https://63014c012a41dd0b1cdb190b--layer5.netlify.app

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js [hunk 1]
```diff
@@ -1,44 +1,173 @@
...
-          ) : (
-            <div>
-              <Link
-                to={`/service-mesh-management/meshery${item.fields.slug}`}
-              >
-                <div>
+          if (status){
+            return (
+              <Hexagon className="container" style={{ background: "#A0AAAA" }}>
+                <div style={{ background: "#A0AAAA" }}>
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:138]: Why do we need this div, again?

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js [hunk 2]
```diff
@@ -1,44 +1,173 @@
...
 
-          return data;
-        })}
-      </div>
+            return (
+              <Hexagon className="container">
+                <Link
+                  to={`/service-mesh-management/meshery${item.fields.slug}`}
+                >
+                  <div>
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:157]: Same as above

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js [hunk 3]
```diff
@@ -1,44 +1,173 @@
...
+      categoryList.forEach((item) => {
+        if (item.name === category) item.isSelected = true;
+      });
+    } else {
+      categoryList[0].isSelected = true;
+    }
+
+    setCategoryList(categoryList);
+    setIntegrationCollection();
+
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:55]: Remove such extra line spacings from the code

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 3 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:55]: I added extra line spacing to make code more readable. :)

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 1 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:138]: We need this div to make the border hover effect.

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 3 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:55]: That's fine, but in most places, it is redundant. Like no extra line is needed after a block code ends. Remove them

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 1 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:138]: Use the div that is rendered by the `HoneyComb` component. No need for an extra div.
And, if that doesn't work, use `::after` to do so.

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 3 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:55]: Okay!

**Comment by l5io** [comment]
🚀 Preview for commit 4f7c31149a7e892f7910aa1b31037d88af6995c7 at: https://6301fd496134776d28c38928--layer5.netlify.app

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 1 above)
Comment (leecalcote) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js]: Thank you. Too many layers. The on hover **border** effect should not be made through another div, but... yes, that's right: **borders**.

Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js]: I removed that extra div in the lastest commit.

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js [hunk 4]
```diff
@@ -1,44 +1,147 @@
...
+    { id: 4, name: "Operating System", isSelected: false },
+    { id: 5, name: "Collaboration", isSelected: false },
+  ]);
+
+  useEffect(() => setCategory(), []);
+
+  const setCategory = () => {
+    if (category !== undefined){
+      categoryList.forEach((item) => {
+        if (item.name === category) item.isSelected = true;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:45]: Please don't like `if` statements like these.
It is very confusing. Update at all the places (part of changes) in this PR.

File: src/sections/Meshery/Meshery-integrations/Integration.style.js [hunk 5]
```diff
@@ -1,93 +1,47 @@
...
-      
-    
-    }
+        ::after{
+          content: "";
+          background: white;
+          margin: 0px 0px 0 6px;
+          clip-path: polygon(50% 0%,100% 25%,100% 75%,50% 100%,0% 75%,0% 25%);
+          height: 93%;
+          width: 93%;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Integration.style.js:22]: Finally, thank you!
The border width needs to be reduced a bit.
Probably by moderating the width, height, left, and top values a bit.

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 4 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:45]: I didn't get you. What should I do if you don't want if statements like this?

File: src/sections/Meshery/Meshery-integrations/Integration.style.js (hunk 5 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/Integration.style.js:22]: Okay!

File: src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js (hunk 4 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:45]: Convert to blocks.
Sinply write like
```
if (condition) {
        statements
}
```

Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:45]: Okay

**Comment by l5io** [comment]
🚀 Preview for commit ec2be1fe3c4f5cb8571cae095445a34316af915e at: https://63028a645f387959fc891eee--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 91ede0165c37639fbd00c56687bee73345c751f9 at: https://63035ba9ab4f3a633ae3712c--layer5.netlify.app

**Review by Nikhil-Ladha — APPROVED** [review]

**Pull request merged**